src/ion/merge.rs

```diff
@@ -258,23 +271,30 @@ impl<'a, F: Function> Env<'a, F> {
             }
 
             let mut fixed = false;
+            let mut fixed_def = false;
             let mut stack = false;
             for entry in &self.bundles[bundle.index()].ranges {
                 for u in &self.ranges[entry.index.index()].uses {
                     if let OperandConstraint::FixedReg(_) = u.operand.constraint() {
                         fixed = true;
+                        if u.operand.kind() == OperandKind::Def {
+                            fixed_def = true;
+                        }
                     }
                     if let OperandConstraint::Stack = u.operand.constraint() {
                         stack = true;
                     }
-                    if fixed && stack {
+                    if fixed && stack && fixed_def {
                         break;
                     }
                 }
             }
             if fixed {
                 self.bundles[bundle.index()].set_cached_fixed();
             }
+            if fixed_def {
+                self.bundles[bundle.index()].set_cached_fixed_def();
```

Comment:
Does this flag ever get cleared, or does it follow the parts of the bundle if it's split up?

Reply:
Oh this isn't a flag on the whole bundle, just that range. Never mind :)

Reply:
It's recomputed along with the other flags when the bundle is split, if it ever is (it actually is a flag on the whole bundle!).

Reply:
https://github.com/cfallin/regalloc2/blob/ac8e5f4e22791e7e554c2c8f2ddea28768bbdc78/src/ion/process.rs#L344-L350